src/emailmessagelistmodel.cpp

```diff
@@ -924,3 +932,51 @@ void EmailMessageListModel::messagesRemoved(const QMailMessageIdList &ids)
         }
     }
 }
+
+void EmailMessageListModel::searchOnline()
+{
+    // Check if the search term did not change yet,
+    // if changed we skip online search until local search returns again
+    if (m_remoteSearch == m_search) {
+        qCDebug(lcGeneral) << "Starting remote search for " << m_search;
+        EmailAgent::instance()->searchMessages(m_searchKey, m_search, QMailSearchAction::Remote, 100);
+    }
+
+}
+
+void EmailMessageListModel::onSearchCompleted(const QString &search, const QMailMessageIdList &matchedIds,
+                                              bool isRemote, int remainingMessagesOnRemote, EmailAgent::SearchStatus status)
+{
+    if (m_search.isEmpty()) {
+        return;
+    }
+
+    if (search != m_search) {
+        qCDebug(lcGeneral) << "Search terms are different, skipping. Received: " << search << " Have: " << m_search;
+        return;
+    }
+    switch (status) {
+    case EmailAgent::SearchDone:
+        if (isRemote) {
+            // Append online search results to local ones
+            setKey(key() | QMailMessageKey::id(matchedIds));
+            qCDebug(lcGeneral) << "We have more messages on remote " << remainingMessagesOnRemote;
```

Comment:
remainingMessagesOnRemote wil be added as a property here, then users of the model can show a get more/search more on server.